src/dev/code_coverage/ingest_coverage/transforms.js

```diff
@@ -83,20 +91,63 @@ export const staticSite = (urlBase) => (obj) => {
   return { ...obj, staticSiteUrl: prokForBoth() };
 };
 
+const leadingSlashRe = /^\//;
+export const maybeDropLeadingSlash = (x) =>
+  leadingSlashRe.test(x) ? Either.right(x) : Either.left(x);
+export const dropLeadingSlash = (x) => x.replace(leadingSlashRe, '');
+export const stripLeading = (x) => maybeDropLeadingSlash(x).fold(id, dropLeadingSlash);
+
 export const coveredFilePath = (obj) => {
   const { staticSiteUrl, COVERAGE_INGESTION_KIBANA_ROOT } = obj;
 
   const withoutCoveredFilePath = always(obj);
-  const leadingSlashRe = /^\//;
-  const maybeDropLeadingSlash = (x) => (leadingSlashRe.test(x) ? Either.right(x) : Either.left(x));
-  const dropLeadingSlash = (x) => x.replace(leadingSlashRe, '');
-  const dropRoot = (root) => (x) =>
-    maybeDropLeadingSlash(x.replace(root, '')).fold(id, dropLeadingSlash);
+  const dropRoot = (root) => (x) => stripLeading(x.replace(root, ''));
   return maybeTotal(staticSiteUrl)
     .map(dropRoot(COVERAGE_INGESTION_KIBANA_ROOT))
     .fold(withoutCoveredFilePath, (coveredFilePath) => ({ ...obj, coveredFilePath }));
 };
 
+const findTeam = (x) => x.match(/.+\s{1,3}(.+)$/, 'gm');
+export const pluckIndex = (idx) => (xs) => xs[idx];
+const pluckTeam = pluckIndex(1);
+
+export const teamAssignment = (teamAssignmentsPath) => (log) => async (obj) => {
+  const { coveredFilePath } = obj;
+  const isTotal = Either.fromNullable(obj.isTotal);
+
+  return isTotal.isRight() ? obj : await assignTeam(teamAssignmentsPath, coveredFilePath, log, obj);
+};
+export const last = (x) => {
+  const xs = x.split('\n');
+  const len = xs.length;
+
+  return len === 1 ? xs[0] : xs[len - 1];
+};
+const logErr = (aPath) => (log) => (e) => {
+  log.error(`\n!!! Unknown Team for path: \n${pink(aPath)}\n!!! Orig Err: \n${pink(e)}`);
+  return e;
+};
+export async function assignTeam(teamAssignmentsPath, coveredFilePath, log, obj) {
+  const params = [coveredFilePath, teamAssignmentsPath];
+  const logGrepFail = logErr(coveredFilePath)(log);
+
+  const grepAndXform = pipe(
+    TE.tryCatch(() => execa('grep', params, { cwd: ROOT_DIR }), pipe(logGrepFail, id)),
+    TE.map(R.prop('stdout')),
+    TE.map(last),
+    TE.map(findTeam),
+    TE.map(pluckTeam)
+  );
+
+  return pipe(
+    await grepAndXform(),
```

Comment:
@rylnd I went ahead with your first proposition.  Honestly, it was simply easier to grok :)